libs/wingc/src/lsp/document_symbols.rs

```diff
@@ -94,10 +95,18 @@ pub fn on_document_symbols(params: lsp_types::DocumentSymbolParams) -> Vec<Docum
 		let mut visitor = DocumentSymbolVisitor::new();
 		visitor.visit_scope(scope);
 
-		visitor.document_symbols
+		visitor
+			.document_symbols
+			.into_iter()
+			.filter(|sym| filter_symbol(sym))
+			.collect()
 	})
 }
 
+fn filter_symbol(symbol: &DocumentSymbol) -> bool {
+	!{ symbol.name.starts_with(CLOSURE_CLASS_PREFIX) || symbol.name.starts_with(PARENT_THIS_NAME) }
+}
```

Comment:
Would it make sense to add a field on the `Symbol` struct that indicates if a symbol is generated or not?

Reply:
Maybe, although it maybe belongs on the span instead (since we also generate things other than symbols).

I suspect we'll have to tackle this when we start dealing with multiple files.